src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -481,4 +613,72 @@ void MatterAccessControlPluginServerInitCallback()
 
     AttributeAccessInterfaceRegistry::Instance().Register(&sAttribute);
     GetAccessControl().AddEntryListener(sAttribute);
+
+#if CHIP_CONFIG_USE_ACCESS_RESTRICTIONS
+    auto accessRestrictionProvider = GetAccessControl().GetAccessRestrictionProvider();
+    if (accessRestrictionProvider != nullptr)
+    {
+        accessRestrictionProvider->AddListener(sAttribute);
+    }
+#endif
+}
+
+#if CHIP_CONFIG_USE_ACCESS_RESTRICTIONS
+bool emberAfAccessControlClusterReviewFabricRestrictionsCallback(
+    CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+    const Clusters::AccessControl::Commands::ReviewFabricRestrictions::DecodableType & commandData)
+{
+    if (commandPath.mEndpointId != 0)
+    {
+        ChipLogError(DataManagement, "AccessControlCluster: invalid endpoint in ReviewFabricRestrictions request");
+        return true;
+    }
+
+    uint64_t token;
+    std::vector<AccessRestrictionProvider::Entry> entries;
+    auto entryIter = commandData.arl.begin();
+    while (entryIter.Next())
+    {
+        AccessRestrictionProvider::Entry entry;
+        entry.fabricIndex    = commandObj->GetAccessingFabricIndex();
+        entry.endpointNumber = entryIter.GetValue().endpoint;
+        entry.clusterId      = entryIter.GetValue().cluster;
+
```

Comment:
I don't think this is possible/feasible as these entries may be different than those currently in the ARL.

The spec must be updated so that each ARL entry has a unique handle, and the request has a list of handles that are to be reviewed. Otherwise it's extremely error prone and it's possible the entries cannot be reviewed or cryptic errors will arise.

Reply:
Tracked in https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/10338